File: backend/app/Models/QuestionStat.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class QuestionStat extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'topic_id',
        'total_questions',
        'correct',
        'wrong',
        'empty',
    ];

    protected function casts(): array
    {
        return [
            'total_questions' => 'integer',
            'correct'         => 'integer',
            'wrong'           => 'integer',
            'empty'           => 'integer',
        ];
    }

    public function topic(): BelongsTo
    {
        return $this->belongsTo(Topic::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: mobile/native-ios/EduMindAI/Models/QuestionStat.swift
import Foundation

struct QuestionStat: Decodable, Identifiable {
    let id: Int
    let userId: Int?
    let topicId: Int?
    let totalQuestions: Int
    let correct: Int
    let wrong: Int
    let empty: Int
    let createdAt: String?
    let updatedAt: String?
    let topic: QuestionStatTopic?
}

struct QuestionStatTopic: Decodable, Identifiable {
    let id: Int
    let name: String
    let subjectId: Int?
    let subject: StudySubject?
}

struct QuestionStatFormPayload: Encodable {
    let totalQuestions: Int
    let correct: Int
    let wrong: Int
    let empty: Int
}
